scipy/stats/morestats.py

```diff
@@ -3147,16 +3147,32 @@ def median_test(*args, **kwds):
     return stat, p, grand_median, table
 
 
-def _circfuncs_common(samples, high, low):
+def _circfuncs_common(samples, high, low, **kwds):
+    # Ensure samples are array-like
     samples = np.asarray(samples)
+
+    # Read and apply the NaN policy, defaulting to propagate (previous behavior)
+    nan_policy = kwds.pop('nan_policy', 'propagate')
+    if len(kwds) > 0:
+        raise TypeError("unexpected keyword argument %r" % kwds.keys()[0])
+
+    contains_nan, nan_policy = _contains_nan(samples, nan_policy)
+    if contains_nan:
+        if nan_policy == 'propagate':
+            return np.nan, np.nan
```

Comment:
I know we do this in several other functions, but I think this is wrong for functions that have an `axis` keyword - it will change the output shape from an array to a scalar. If you add a test for, e.g., a 2-D array with `axis=0` or `axis=1` you should see this.